src/data-types/nvarchar.ts

```diff
@@ -29,23 +33,24 @@ module.exports = {
   },
 
   resolveLength: function(parameter) {
+    const value = parameter.value as any; // Temporary solution. Remove 'any' later.
     if (parameter.length != null) {
       return parameter.length;
-    } else if (parameter.value != null) {
-      if (Buffer.isBuffer(parameter.value)) {
-        return (parameter.value.length / 2) || 1;
+    } else if (value != null) {
+      if (Buffer.isBuffer(value)) {
+        return (value.length / 2) || 1;
       } else {
-        return parameter.value.toString().length || 1;
+        return value.toString().length || 1;
       }
     } else {
       return this.maximumLength;
     }
   },
 
-  writeTypeInfo: function(buffer, parameter) {
+  writeTypeInfo: function(buffer, parameter:ParameterData<any>) {
```

Comment:
I think you don't have to specify `ParameterData<any>` here (and in other places) anymore:

```suggestion
  writeTypeInfo: function(buffer, parameter) {
```

Reply:
Should we keep the parameter:ParameterData<any> for the places that use "parameter.value.length"? Otherwise, the parameter. the value here will be unknown, and it will complain about calling length based on unknow. 

or we should do: 
const value =parameter.value as any.  instead?